src/dataprotection/azext_dataprotection/manual/_params.py

```diff
@@ -114,6 +114,18 @@ def load_arguments(self, _):
         c.argument('secret_store_uri', type=str, help="specify the secret store uri to use for authentication")
         c.argument('snapshot_resource_group_name', options_list=['--snapshot-resource-group-name', '--snapshot-rg'], type=str, help="Name of the resource group in which the backup snapshots should be stored")
         c.argument('tags', tags_type)
+        c.argument('use_system_assigned_identity', options_list=['--use-system-identity', '--use-system-assigned-identity'], arg_type=get_three_state_flag(), help="Use system assigned identity")
```

Comment:
Seems not following managed identity design guideline https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md

Reply:
cc @zhoxing-ms for review

Reply:
Is it sufficient to update the parameters in `initialize` commands with `--mi-system-assigned` and `--mi-user-assigned`, and in the update commands with `--system-assigned`/`--user-assigned`?

in `backup-vault create/update` we already set the managed identity type with the `--type` parameter and the UAMI URLs with `--uami`

![image](https://github.com/user-attachments/assets/3bd435fa-093f-4f3c-9cd8-7723c3a4b3da)

Reply:
Then it's inconsistency between dataprotection module and other modules because many modules followed the design guideline.

Reply:
Actually, we suggest the entire module referring to this design guideline https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md, but there were other commands in the module that did not refer to it before, right?

If so, may I ask if migrating the previously inconsistent commands to the design guideline through the breaking change process (add deprecation info for the original old parameters `--uami` and redirect it to the new parameters wich are in line with our design guideline) would be a heavy workload?

Reply:
I'll explore the possibility of migrating the previously inconsistent commands. However, if the change is significantly heavy and does not fit in my current time commitments, what is the plan of action - can I just make the change to the commands being currently added?

Reply:
May I ask if there were many inconsistent commands before, and if the experience of commands within the `dataprotection` extension is inconsistent, is it acceptable for your customers?

Reply:
Hi @zhoxing-ms, I was able to make the change across old commands as well.

One caveat - with `az dataprotection backup-vault identity remove`, if I have multiple UAMIs associated but I try to remove one specific value using the `--user-assigned` option as described in the [command guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md#operate-managed-identity-on-existing-resource), then this is ignored, and all the associated UAMIs get removed. Is this acceptable behaviour?

If not, what should we do - this is code autogenerated using AAZ-dev-tools, and I'm not sure what change to make to accomplish otherwise.

Reply:
> but I try to remove one specific value using the --user-assigned option as described in the [command guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md#operate-managed-identity-on-existing-resource), then this is ignored, and all the associated UAMIs get removed. Is this acceptable behaviour?

May I ask why not using this `az <resource> identity remove ... --system-assigned --user-assigned <AzureResourceId1>` to remove the specific user assigned identity?

Reply:
> > but I try to remove one specific value using the --user-assigned option as described in the [command guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md#operate-managed-identity-on-existing-resource), then this is ignored, and all the associated UAMIs get removed. Is this acceptable behaviour?
> 
> May I ask why not using this `az <resource> identity remove ... --user-assigned <AzureResourceId1>` command to remove the specific user assigned identity?

It isn't working, even if I specify a user assigned identity to remove, it removes all of them.

Reply:
May I ask what the reason is? This sounds like a code bug rather than a parameter design issue

Reply:
Hey @zhoxing-ms I've fixed the bug. It behaves as expected now